File: qc-core/src/main/java/com/eager/questioncloud/core/domain/creator/infrastructure/repository/CreatorRepositoryImpl.java
package com.eager.questioncloud.core.domain.creator.infrastructure.repository;

import static com.eager.questioncloud.core.domain.creator.infrastructure.entity.QCreatorEntity.creatorEntity;
import static com.eager.questioncloud.core.domain.creator.infrastructure.entity.QCreatorStatisticsEntity.creatorStatisticsEntity;
import static com.eager.questioncloud.core.domain.user.infrastructure.entity.QUserEntity.userEntity;

import com.eager.questioncloud.core.domain.creator.dto.CreatorInformation;
import com.eager.questioncloud.core.domain.creator.infrastructure.entity.CreatorEntity;
import com.eager.questioncloud.core.domain.creator.model.Creator;
import com.eager.questioncloud.core.exception.CoreException;
import com.eager.questioncloud.core.exception.Error;
import com.querydsl.core.types.Projections;
import com.querydsl.jpa.impl.JPAQueryFactory;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;

@Repository
@RequiredArgsConstructor
public class CreatorRepositoryImpl implements CreatorRepository {
    private final CreatorJpaRepository creatorJpaRepository;
    private final JPAQueryFactory jpaQueryFactory;

    @Override
    public Boolean existsById(Long creatorId) {
        return creatorJpaRepository.existsById(creatorId);
    }

    @Override
    public CreatorInformation getCreatorInformation(Long creatorId) {
        CreatorInformation creatorInformation = jpaQueryFactory.select(Projections.constructor(
                CreatorInformation.class,
                creatorEntity.id,
                userEntity.userInformationEntity.name,
                userEntity.userInformationEntity.profileImage,
                creatorEntity.creatorProfileEntity.mainSubject,
                userEntity.userInformationEntity.email,
                creatorStatisticsEntity.salesCount,
                creatorStatisticsEntity.averageRateOfReview,
                creatorEntity.creatorProfileEntity.introduction))
            .from(creatorEntity)
            .where(creatorEntity.id.eq(creatorId))
            .leftJoin(userEntity).on(userEntity.uid.eq(creatorEntity.userId))
            .leftJoin(creatorStatisticsEntity).on(creatorStatisticsEntity.creatorId.eq(creatorEntity.id))
            .fetchFirst();

        if (creatorInformation == null) {
            throw new CoreException(Error.NOT_FOUND);
        }

        return creatorInformation;
    }

    @Override
    public Creator save(Creator creator) {
        return creatorJpaRepository.save(CreatorEntity.from(creator)).toModel();
    }

    @Override
    public void deleteAllInBatch() {
        creatorJpaRepository.deleteAllInBatch();
    }
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/creator/infrastructure/repository/CreatorRepositoryImpl.kt
package com.eager.questioncloud.core.domain.creator.infrastructure.repository

import com.eager.questioncloud.core.domain.creator.infrastructure.entity.CreatorEntity.Companion.from
import com.eager.questioncloud.core.domain.creator.model.Creator
import com.eager.questioncloud.core.exception.CoreException
import com.eager.questioncloud.core.exception.Error
import org.springframework.stereotype.Repository

@Repository
class CreatorRepositoryImpl(
    private val creatorJpaRepository: CreatorJpaRepository,
) : CreatorRepository {

    override fun findById(creatorId: Long): Creator {
        return creatorJpaRepository.findById(creatorId)
            .orElseThrow { CoreException(Error.NOT_FOUND) }
            .toModel()
    }

    override fun findByUserId(userId: Long): Creator? {
        return creatorJpaRepository.findByUserId(userId)?.toModel()
    }

    override fun findByIdIn(creatorIds: List<Long>): List<Creator> {
        return creatorJpaRepository.findByIdIn(creatorIds).map { it.toModel() }
    }

    override fun existsById(creatorId: Long): Boolean {
        return creatorJpaRepository.existsById(creatorId)
    }

    override fun save(creator: Creator): Creator {
        return creatorJpaRepository.save(from(creator)).toModel()
    }

    override fun deleteAllInBatch() {
        creatorJpaRepository.deleteAllInBatch()
    }
}
